File: algo_c/nondec-plus.c
#include <stdio.h>

int ans[30] = {0};

int main() {
    int num1, num2, dig;
    while (scanf("%d%d%d", &num1, &num2, &dig) != EOF) {
        int n = 0;
        int sum = num1 + num2;

        ans[n] = sum % dig;
        sum = sum / dig;

        while (sum != 0) {
            n = n + 1;
            ans[n] = sum % dig;
            sum = sum / dig;
        }

        for (int i = n; i >= 0; i--) {
            printf("%d", ans[i]);
        }
        printf("\n");
    }
}

File: algo-go/nondec-plus.go
package main

import (
    "fmt"
    "io"
)

func main() {
    var ans [30]int
    for {
        var num1, num2, dig int
        _, err := fmt.Scanf(
            "%d %d %d",
            &num1, &num2, &dig,
        )
        if err == io.EOF {
            break
        }

        sum := num1 + num2

        n := 0
        ans[n] = sum % dig
        sum = sum / dig

        for ; sum != 0; {
            n = n + 1
            ans[n] = sum % dig
            sum = sum / dig
        }

        for i := n ; i >= 0; i-- {
            fmt.Printf("%d", ans[i])
        }
        fmt.Printf("\n");
    }
}
